src/event_loop_dispatcher.hpp

```diff
@@ -0,0 +1,134 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2019 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_DART_EVENT_LOOP_DISPATCHER_HPP
+#define REALM_DART_EVENT_LOOP_DISPATCHER_HPP
+
+#include <realm/object-store/util/scheduler.hpp>
+
+#include <tuple>
+#include <iostream>
+#include "dart_api_dl.h"
+
+namespace realm::util {
+
+template <class F>
+class EventLoopDispatcher;
+
+template <typename... Args>
+class EventLoopDispatcher<void(Args...)> {
+    using Tuple = std::tuple<typename std::remove_reference<Args>::type...>;
+
+private:
+    const std::shared_ptr<util::UniqueFunction<void(Args...)>> m_func;
+    std::shared_ptr<util::Scheduler> m_scheduler; //  = util::Scheduler::make_default();
+
+public:
+    EventLoopDispatcher(std::shared_ptr<util::Scheduler> scheduler, util::UniqueFunction<void(Args...)> func)
+        : m_func(std::make_shared<util::UniqueFunction<void(Args...)>>(std::move(func))), m_scheduler(scheduler)
+    {
+    }
+
+    const util::UniqueFunction<void(Args...)>& func() const
+    {
+        return *m_func;
+    }
+
+    void operator()(Args... args)
+    {
+        m_scheduler->invoke(
```

Comment:
I am not sure we are doing it right here. Why we are using (relying on) core scheduler instead of calling our own implementation with Dart_PostInterger. 

At a high level we need to be able to store a function pointer and args in a collection and then iterate the collection invoking the functions on the correct isolate thread. What is the core scheduler helping us with here

Reply:
The `EventLoopDispatcher` can work with any `Scheduler`. We take it as an argument on construction. For our use we will always use the `realm_scheduler_t` created with `realm_dart_create_scheduler` and passed in the call.